tests/driver/cpp/cppdriver.rs

```diff
@@ -141,3 +142,20 @@ pub fn test(testcase: &test_driver_lib::TestCase) -> Result<(), Box<dyn Error>>
 
     Ok(())
 }
+
+fn library_search_path_env_with(
+    value_to_prepend: &str,
+) -> impl IntoIterator<Item = (&'static str, String)> {
+    let (var, separator) = if cfg!(target_os = "windows") {
+        ("PATH", ';')
+    } else if cfg!(target_os = "macos") {
+        ("DYLD_FALLBACK_LIBRARY_PATH", ':')
+    } else {
+        ("LD_LIBRARY_PATH", ':')
+    };
+
+    std::iter::once((
+        var,
+        format!("{}{}{}", value_to_prepend, separator, std::env::var(var).unwrap_or_default()),
```

Comment:
If the env variable was empty, we end up with a trailing separator.  Is that acceptable?

Reply:
Yes, to be best of my knowledge all operating systems ignore that.

Reply:
Linux treats an empty LD_LIBRARY_PATH segment as `.`. So in general it is not OK to have empty path segments in LD_LIBRARY_PATH, becuase it is non-obvious and will load libs from places that are typically under user control.

Not sure how much of an issue that is here though.

Reply:
There are two environment variables here. `CPP_LIB_PATH` and `LD_LIBRARY_PATH`. The former cannot be empty (the rust build fails!), `LD_LIBRARY_PATH` could be empty though. In that case we end up setting it to merely the path  separator, so it will be `LD_LIBRARY_PATH=:` which I think is harmless.